managers/bosh-manager/DirectorService.js

```diff
@@ -0,0 +1,1112 @@
+    logger.debug(`+-> Checking diff of deployment ${deploymentName}`);
+    return this
+      .regenerateManifest(deploymentName, opts)
+      .then(manifest => this.director
+        .diffDeploymentManifest(deploymentName, manifest)
+        .then(result => _.set(result, 'manifest', manifest))
+      );
+  }
+
+  regenerateManifest(deploymentName, opts) {
+    return this.director
+      .getDeploymentManifest(deploymentName)
+      .then(manifest => this.generateManifest(deploymentName, _.extend(opts, {
+        previous_manifest: manifest
+      })));
+  }
+
+  generateManifest(deploymentName, opts, preDeployResponse, preUpdateAgentResponse) {
+    const index = opts.network_index || this.getNetworkSegmentIndex(deploymentName);
+    const networks = this.getNetworks(index);
+    const allRequiredNetworks = _.union(networks.dynamic, networks.all.filter(net => _.startsWith(net.name, this.networkName)));
+    const tags = opts.context;
+    const skipAddOns = _.get(opts, 'skip_addons', false) || _.get(config, 'service_addon_jobs', []).length === 0;
+    const header = new Header({
+      name: deploymentName,
+      director_uuid: this.director.uuid,
+      releases: this.releases,
+      stemcells: [this.stemcell],
+      tags: tags,
+      networks: _.map(allRequiredNetworks, net => net.toJSON()),
+      release_name: !skipAddOns ? config.release_name : undefined,
+      release_version: !skipAddOns ? config.release_version : undefined
+    });
+    const context = new EvaluationContext(_.assign({
+      index: index,
+      header: header,
+      cpi: this.director.cpi,
+      networks: networks[this.networkName],
+      parameters: opts.parameters || {},
+      properties: this.settings.context || {},
+      previous_manifest: opts.previous_manifest,
+      multi_az_enabled: config.multi_az_enabled,
+      stemcell: this.stemcell,
+      actions: preDeployResponse,
+      preUpdateAgentResponse: preUpdateAgentResponse
+    }, opts.context));
+    logger.info('Predeploy response -', preDeployResponse);
+    if (networks[this.networkName] === undefined) {
+      logger.error(`subnet ${this.networkName} definition not found among the applicable networks defintion : ${JSON.stringify(networks)}`);
+      throw new errors.UnprocessableEntity(`subnet ${this.networkName} definition not found`);
+    }
+    let manifestYml = _.template(this.template)(context);
+    if (!skipAddOns) {
+      const serviceManifest = yaml.safeLoad(manifestYml);
+      this.configureAddOnJobs(serviceManifest, context.spec);
+      manifestYml = yaml.safeDump(serviceManifest);
+    }
+    return manifestYml;
+  }
+
+  configureAddOnJobs(serviceManifest, context) {
+    const addOns = new Addons(context).getAll();
+    if (serviceManifest.addons) {
+      serviceManifest.addons = serviceManifest.addons.concat.apply(serviceManifest.addons, addOns);
+    } else {
+      serviceManifest.addons = addOns;
+    }
+  }
+
+  // findDeploymentTask(deploymentName) {
```

Comment:
Please do not checkin commented out code. They tend to exist for ever.

Reply:
This methods are copied from old Instance/Manager files. They are still used with the old flow as part of dashboard show, which should eventually be removed. Kept it here so that we can uncomment it here and remove from old Instance/Manager files. Also we decided not to touch the  Instance/Manager files in the current PR. They also will be cleaned up.